### PR TITLE
DLP: Added sample for inspecting data with hotword rule

### PR DESCRIPTION
**Pull request by jainamtrivedi-crest**
- Added sample for [Inspect data with hotword rule](https://cloud.google.com/dlp/docs/creating-custom-infotypes-likelihood.md#match-mrn) and unit tests.

### CODE REVIEW

File: dlp/api/Snippets.Tests/InspectDataWithHotwordRuleTests.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        private DlpTestFixture _fixture;
+        public InspectDataWithHotwordRuleTests(DlpTestFixture fixture) => _fixture = fixture;
+        [Fact]
+        public void TestInspectDataWithHotwordRule()
+        {
+            string text = "Patient's MRN 444-5-22222 and just a number 333-2-33331233";
+            string customRegexPattern = "[1-9]{3}-[1-9]{1}-[1-9]{5}";
+            string hotwordRegexPattern = "(?i)(mrn|medical)(?-i)";
+            var result = InspectDataWithHotwordRule.InspectDataHotwordRule(_fixture.ProjectId, text, customRegexPattern, hotwordRegexPattern);
+            Assert.Equal(2, result.Result.Findings.Count);
```
Comment (annadata) [inline dlp/api/Snippets.Tests/InspectDataWithHotwordRuleTests.cs:30]: Please add an assert to confirm "Info type: C_MRN" is present in the output. Or equivalent assert in the "result"

Comment (jainamtrivedi-crest) [reply dlp/api/Snippets.Tests/InspectDataWithHotwordRuleTests.cs:30]: Added.

File: dlp/api/Snippets/InspectDataWithHotwordRule.cs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        {
+            ByteItem = new ByteContentItem
+            {
+                Type = ByteContentItem.Types.BytesType.TextUtf8,
+                Data = Google.Protobuf.ByteString.CopyFromUtf8(text)
+            }
+        };
+
+        // Construct the info type.
+        var infoType = new InfoType { Name = "C_MRN" };
```
Comment (annadata) [inline dlp/api/Snippets/InspectDataWithHotwordRule.cs:44]: I suggest parameterising the infoType name as well. That way the entire custom info type including its name will be controlled by the caller.

**Review by annadata — APPROVED** [review]

File: dlp/api/Snippets/InspectDataWithHotwordRule.cs (hunk 2 above)
Comment (jainamtrivedi-crest) [reply dlp/api/Snippets/InspectDataWithHotwordRule.cs]: Done.

**Review by amanda-tarafa — APPROVED** [review]

**Pull request merged**